docs/rag/ilab-rag-retrieval.md

```diff
@@ -0,0 +1,437 @@
+
+**Notes**:
+
+* To ensure consistency and avoid issues with document versioning or outdated embeddings, the ingested collection will be cleared before execution.
+  This ensures it contains only the embeddings generated from the most recent run.
+
+### Ingestion-Why We Need It
+
+To populate embedding vector stores with pre-processed information that can be used at chat inference time.
+
+#### Ingestion-Supported Databases
+
+The command may support various vector database types. A default configuration will align with the selected
+InstructLab technology stack.
+
+#### Ingestion-Usage
+
+The generated embeddings can later be retrieved from a vector database and converted to text, enriching the
+context for RAG-based chat pipelines.
+
+### 2.5 Embedding Ingestion Pipeline Options
+
+**Note**: The `--help` option will be aware of the `rag` command group only if `ILAB_DEV_PREVIEW` environment variable is set to `true`.
+
+```bash
+% ilab rag ingest --help 
+Usage: ilab rag ingest [OPTIONS]
+
+  The embedding ingestion pipeline
+
+Options:
+  --document-store-type TEXT      The document store type, one of:
+                                  `milvuslite`.
+  --document-store-uri TEXT       The document store URI
+  --document-store-collection-name TEXT
+                                  The document store collection name
+  --model-dir TEXT                Base directories where models are stored.
+                                  [default: (The default system model location
+                                  store, located in the data directory.)]
+  --embedding-model TEXT          The embedding model name
+  --output-dir TEXT               Directory where generated datasets are
+                                  stored.
+  --input DIRECTORY               Directory where pre-processed documents are
+                                  located.
+  --help                          Show this message and exit.
+```
+
+| Option Description | Default Value | CLI Flag | Environment Variable |
+|--------------------|---------------|----------|----------------------|
+| Document store implementation, one of: `milvuslite`, **TBD** | `milvuslite` | `--document-store-type` | `ILAB_DOCUMENT_STORE_TYPE` |
+| Document store service URI. | `./embeddings.db` | `--document-store-uri` | `ILAB_DOCUMENT_STORE_URI` |
+| Document store collection name. | `IlabEmbeddings` | `--document-store-collection-name` | `ILAB_DOCUMENT_STORE_COLLECTION_NAME` |
+| Base directories where models are stored. | `$HOME/.cache/instructlab/models`  | `--retriever-embedder-model-dir` | `ILAB_EMBEDDER_MODEL_DIR` |
+| Name of the embedding model. | **TBD** | `--retriever-embedder-model-name` | `ILAB_EMBEDDER_MODEL_NAME` |
+
+### 2.6 RAG Chat Pipeline Command
+
+The proposal is to add a `chat.rag.enable` configuration (or the equivalent `--rag` flag) to the `model chat` command, like:
```

Comment:
This is also gated on ILAB_DEV_PREVIEW, right?  If so, let's make that explicit here.

Reply:
+1